auth/saml/index.php

```diff
@@ -9,11 +9,25 @@
         // We read saml parameters from a config file instead from the database
         // due we can not operate with the moodle database without load all
         // moodle session issue.
-        if(!file_exists('saml_config.php')) {
+				//We need to get dataroot from CFG without loading moodle database
+
+				$config = file_get_contents('../../config.php');
+
+				$config = preg_replace('/\s+/', '', $config);
+				if(preg_match('/^.+?\$CFG\-\>dataroot=[\'"](.+?)[\'"];/', $config, $match)){
```

Comment:
First time this file does not exists, why not create it with default values instead? 

Also in order to apply this patch I need that old version of the plugin keep working, so if someone already saved the file on the saml plugin folder, we need to read the content and save it in the moodledata folder.